viewing_party/party.py

```diff
@@ -1,23 +1,286 @@
-# ------------- WAVE 5 --------------------
-# -----------------------------------------
+    for movie in user_data["watched"]:
+        if movie["title"] in only_user_watched_set:
+            only_user_watched.append(movie)
+
+    return only_user_watched
+
+
+def get_friends_unique_watched(user_data):
+    """Determine which movies at least one of the user's friends have watched,
+        but the user has not watched
+    
+    Parameters: 
+        user_data: dict, 
+            {
+                "watched": [{}, {}, {}], 
+                "friends": ["watched": [{"title": }, {}, {}]}, {"watched": [{}, {}, {}]}, {"watched": [[{}, {}, {}]}]
+            }
+    Returns: 
+        list: [{}, {}, {}] Return a list of dictionaries, that represents a list of movies
+    """
+    user_movies_list = set()
+    friend_movies_list = set()
+    only_friends_watched = []
+
+    for movie in user_data["watched"]:
+        user_movies_list.add(movie["title"])
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            friend_movies_list.add(friend_movie["title"])
+    only_friends_watched_set = friend_movies_list.difference(user_movies_list)
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if friend_movie["title"] in only_friends_watched_set and friend_movie not in only_friends_watched:
+                only_friends_watched.append(friend_movie)
+
+    return only_friends_watched
+
+
+def get_available_recs(user_data):
+    """Create a list of recommended movies for a user based on their subscriptions and friends' watched movies.
+
+    Parameters:
+        user_data (dict): 
+            A dictionary containing:
+            - "watched": list of movies the user has already watched
+            - "friends": list of friends, each with their own "watched" list
+            - "subscriptions": list of streaming services the user is subscribed to
+
+    Returns:
+        list: Movies that the user hasn't watched, that at least one friend has watched,
+            and are hosted on a service in the user's subscriptions.
+    """
+    
+    recommended_movies = []
+    
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                if movie not in recommended_movies:
+                    recommended_movies.append(movie)
+                        
+    return recommended_movies
+
+
+def get_new_rec_by_genre(user_data):
+    """Determine a list of recommended movies
+
+    Parameters:
+        user_data (dict): 
+            A dictionary containing:
+            - "watched": list of movies the user has already watched
+            - "friends": list of friends, each with their own "watched" list
+
+    Returns:
+        list: list of recommended movies
+    """
+
+    most_watched_genre = get_most_watched_genre(user_data)

```

Comment:
Nice, I like that you invoked a function you already wrote instead of repeating logic here.